### PR TITLE
Feature/223 naming convention for canvas operators

### PR DESCRIPTION
**Pull request by LesterThomas**
This is implementing #223  
and creating the foundation for #364 

I have refactored the component-operator and api-operator-istio into separate container images. There is new documentation at /operators, /operators/component-management and /operators/api-management

Still to do:

Apply same approach to api-operator-kong and api-operator-apisix
Separate identity-config-operator from component-operator (requires new IdentityConfig CRD)

### CODE REVIEW

**Comment by LesterThomas** [comment]
@ferenc-hechler - this PR isn't triggering any checks - I think it should:

It has an updated version of `canvas-oda` as well as `component-operator` and `api-operator-istio` charts.
It includes a preReleaseSuffix for `component-operator` and `api-operator-istio`.

I have integrated the latest **main** branch into this feature branch. Do you know why it is not triggering any checks?

**Comment by ferenc-hechler** [comment]
> @ferenc-hechler - this PR isn't triggering any checks - I think it should:

You refactored the source folders. The dockerbuild-config.yaml contains the paths to the folders, which trigger the rebuild:

https://github.com/tmforum-oda/oda-canvas/blob/21ea78165ed3417557beda6bba8b5a180ecae6a8/automation/generators/dockerbuild-workflow-generator/dockerbuild-config.yaml#L20-L36

The paths have to be updated in the config and afterwards the workflow files have to be regenerated by executing the python script as described in "How to add a new Dockerimage":
https://github.com/tmforum-oda/oda-canvas/blob/feature/223-naming-convention-for-canvas-operators/docs/developer/work-with-dockerimages.md#how-to-add-a-new-dockerimage

We can do this together tomorrow in the clinic session.

**Comment by ferenc-hechler** [comment]
But as there were no checks triggered at all, maybe it has to do with the conflicts.

**Comment by LesterThomas** [comment]
> > @ferenc-hechler - this PR isn't triggering any checks - I think it should:
> 
> You refactored the source folders and updated the paths in the dockerbuild-config.yaml.
> 
> Also you regenerated the workflow files from the templates by running the Python script. I think the issue is, that you did not use the "recurse into subfolders" pattern in the paths:
> 
> https://github.com/tmforum-oda/oda-canvas/blob/21ea78165ed3417557beda6bba8b5a180ecae6a8/automation/generators/dockerbuild-workflow-generator/dockerbuild-config.yaml#L29-L31
> 
> This only triggers on changes in the folder itself. To trigger also for changes in the subfolders you should use ".../**/*":
> 
> ```
>   paths:
>     - source/operators/component-management/**/*
>     - source/operators/identity-config/keycloak/**/*
> ```

The Docker release images are being created successfully (see https://github.com/tmforum-oda/oda-canvas/actions/runs/11799732445). I didn't recursively go through the folders, as these operators only have a single  folder (the subfolders are only for the sequence diagrams).

The issue is that the checks (for linting and testing a PR) are not firing.

**Comment by LesterThomas** [comment]
> is not firing, even though I integrated the main branch into my feature branch and this PR is to pull into main!

I fixed the merge conflicts and now the checks are working - maybe GitHub doesn't trigger actions for PR's with merge conflicts?

**Comment by ferenc-hechler** [comment]
I tested the fresh installation of the canvas from the branch, deployed a component with secrets management enabled and upgraded the canvas deployment successfully (see https://github.com/tmforum-oda/oda-canvas/issues/360#issuecomment-2474091846 )

The first time I had an issue, that the component CR was stuck in "In-Progress-CompCon" and the ExposedAPI CRs did not get into state "ready". I restarted the api-operator-istio, then it progressed into "Complete".

I deinstalled and reinstalled the component, then it was some time (1-2 minutes) in state "In-Progress-SecretMan", but progressed to "Complete". In the logs of both Operators I could find error messages about concurrent modifications. So, maybe both (SecretsManagementOperator & API-Operator-Istio) tried to write back the status to the component CR and the slower one failed each time. 

Both operators have a retry=5, so they should recover after some time. 

I think this is independent of the refactoring and should be analyzed as a seperate issue.

From my side the PR looks good.

File: source/operators/api-management/istio/apiOperatorIstio.py [hunk 1]
```diff
@@ -0,0 +1,1488 @@
...
+import re
+
+logging_level = os.environ.get("LOGGING", logging.INFO)
+
+kopf_logger = logging.getLogger()
+kopf_logger.setLevel(logging.WARNING)
+
+logger = logging.getLogger("APIOperator")
+logger.setLevel(int(logging_level))
+logger.info(f"Logging set to %s", logging_level)
```
[the remark this answers is not in the document]
Comment (LesterThomas) [reply source/operators/api-management/istio/apiOperatorIstio.py:29]: Good idea - done!

File: source/operators/api-management/istio/apiOperatorIstio.py [hunk 2]
```diff
@@ -0,0 +1,1488 @@
...
+    if re.match("^[0-9]+[.][0-9]+[.][0-9]+[.][0-9]+$", APIOPERATORISTIO_PUBLICHOSTNAME):
+        publichostname_loadBalancer = {
+            "ingress": [{"ip": APIOPERATORISTIO_PUBLICHOSTNAME}]
+        }
+    else:
+        publichostname_loadBalancer = {
+            "ingress": [{"hostname": APIOPERATORISTIO_PUBLICHOSTNAME}]
+        }
+
+
```
[the remark this answers is not in the document]
Comment (LesterThomas) [reply source/operators/api-management/istio/apiOperatorIstio.py:75]: Done!

File: charts/component-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+              name: component-operator-secret
+        ports:
+        - name: seccon
+          containerPort: 5000
+      volumes:
+      - name: config-volume
+        configMap:
+          name: operator-command
+          defaultMode: 420
+          optional: true
```
[the remark this answers is not in the document]
Comment (LesterThomas) [reply charts/component-operator/templates/deployment.yaml]: Removed from component-operator and api-operator-istio

**Pull request merged**